src/pkg/utils/utils.go

```diff
@@ -119,6 +119,68 @@ func ExtractJSON(j any, expectedFilepath string) archives.FileHandler {
 	}
 }
 
+// ExtractBytes returns an archives.FileHandler that extracts a byte contents of a file from an archive
+func ExtractBytes(b *[]byte, expectedFilepath string) archives.FileHandler {
+	return func(_ context.Context, file archives.FileInfo) error {
+		if file.NameInArchive != expectedFilepath {
+			return nil
+		}
+
+		stream, err := file.Open()
+		if err != nil {
+			return err
+		}
+		defer stream.Close()
+
+		fileBytes, err := io.ReadAll(stream)
+		if err != nil {
+			return err
+		}
+
+		*b = fileBytes
+		return nil
+	}
+}
+
+func ExtractAllFiles(outDirPath string) archives.FileHandler {
+	return ExtractFile("", outDirPath)
+}
+
+// ExtractFile returns an archives.FileHandler that extracts a file from an archive
+// NOTE: if expectedFilepath is an emtpy string, all files within the archive will be extracted
+func ExtractFile(expectedFilepath string, outDirPath string) archives.FileHandler {
```

Comment:
@catsby do you have opinions on the `expectedFilepath` param having a behavior of "extract everything" if the input is an empty string?